### ISSUE TITLE
FixedBudgetFeatureLearner's top-k utility still averages inactive-task zero placeholders (the #275 fix landed only in the interaction learner)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#379]
## Summary

#275 / PR #276 fixed `FixedBudgetInteractionLearner`'s top-k utility so that, under `utility_task_balancing in {"active", "active_inverse_frequency"}`, inactive-task zero placeholders no longer dilute the top-k mean (mask inactive rows to `-inf`, divide by the selected active count). PR #276 touched only `interaction_features.py`; the byte-identical code in `FixedBudgetFeatureLearner` (`alberta_framework/core/feature_discovery.py`, `_output_utility_signal` and `_aggregate_task_feature_signal`) was left behind:

```python
if self._utility_aggregation == "topk":
    k = min(self._utility_top_k, self._n_tasks)
    return jnp.mean(jnp.sort(weighted_activity, axis=0)[-k:, :], axis=0)   # zeros of inactive tasks averaged in
```

The dilution factor is `(#active heads at that step) / k`, so a feature measured on a sparse-label step is systematically penalised against an identical feature measured on a dense step; `worst_active = argmin(active_scores)` then discards the wrong slot, and the same diluted utilities feed `should_promote`, the promotion pressure, and metrics 2/3/4.

## Falsifier (deterministic, pristine `main` @ `45279f3`)

```python
import jax.numpy as jnp
from alberta_framework import FixedBudgetFeatureLearner, FixedBudgetInteractionLearner
active = jnp.asarray([True, False, False, False]); ema = jnp.ones(4)
out_w = jnp.asarray([[6.0], [5.0], [5.0], [5.0]]); feats = jnp.ones(1)
for k in (1, 2, 4):
    fd = FixedBudgetFeatureLearner(n_features=1, n_tasks=4, candidate_count=0,
                                   utility_aggregation="topk", utility_top_k=k, utility_task_balancing="active")
    print("top_k", k, "feature_discovery:", fd._output_utility_signal(out_w, feats, active, ema), "expected [6.]")
```

Output on `main`:

```
top_k 1 feature_discovery: [6.]  expected [6.]
top_k 2 feature_discovery: [3.]  expected [6.]
top_k 4 feature_discovery: [1.5] expected [6.]
```

Ranking flip (3 heads, `top_k=2`, `balancing="active"`): feature A (1 active head, real signal 5.0) scores `2.5`, feature B (2 active heads, real 3.0) scores `3.0` — the discovery learner declares A the worse feature while the fixed interaction learner (5.0 vs 3.0) declares B. Through the public `update()` with only head 0 observed, utilities shrink by 0.55–0.88× when `top_k` goes 1→2 and 0.40–0.85× at `top_k=3` on the same signal.

Expected: the same semantics as #276 — under task balancing only active rows compete for the top-k and the mean divides by `min(#active, k)`; the `"none"` branch stays byte-identical.

## Dedup

#275 / PR #276 (same defect, `interaction_features.py` only — verified via `gh pr view 276 --json files`). No issue mentions `FixedBudgetFeatureLearner` or `feature_discovery` top-k; no open PR touches `feature_discovery.py`. (`core/interaction_features.py` is a registered evidence source; `core/feature_discovery.py` is not.)

Fix PR follows (failing-test-first).
